File: InsertionSort/InsertionSort.pl
#!/usr/bin/perl

sub swap {
	my $tmp = $inputArray[$_[1]];
	$inputArray[$_[1]] = $inputArray[$_[0]];
	$inputArray[$_[0]] = $tmp;
}

sub InsertionSort {
	for (my $i = 0; $i <= @inputArray-1; $i++) {
		my $j = $i;
		while ($j > 0 && $inputArray[$j-1] > $inputArray[$j]) {
			swap($j-1,$j);
			$j = $j-1;
		}
	}
}

my $str = <STDIN>;
chomp $str;
@inputArray = split(' ', $str);
InsertionSort(0,@inputArray-1);
print("@inputArray\n");

File: InsertionSort/InsertionSort.rb
#!/usr/bin/ruby

def swap(array, i, j)
	tmp = array[j]
	array[j] = array[i]
	array[i] = tmp
end

def InsertionSort(array)
	for i in 0..array.length-1
		j = i
		while j > 0 && array[j-1] > array[j]
			swap(array,j-1,j)
			j = j-1
		end
	end
end

s = gets.chomp
inputArray = s.split(' ').map(&:to_i)
InsertionSort(inputArray)
puts inputArray.join(" ")